### PR TITLE
Optimize

### PR DESCRIPTION
**Pull request by k7d**
Convert Clojure logic to optimized Java logic. On average 50x faster.

### CODE REVIEW

**Comment by zarlen** [comment]
Whoa, Java from the @k7d?

File: src/camelsnake/Converter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * @param sep Separator that will be inserted between parts. Null for none.
+     * @param capitalizeFirst Whether to capitalize first letter of first part.
+     * @param capitalizeRest Whether to capitalize first letter of rest of the
+     *                       parts.
+     * @return Converted string.
+     */
+    static public String convert(String in, String sep,
+                                  boolean capitalizeFirst,
+                                  boolean capitalizeRest) {
+        StringBuffer out = new StringBuffer();
```
Comment (AltSysrq) [inline src/camelsnake/Converter.java]: `StringBuilder`

Comment (AltSysrq) [inline src/camelsnake/Converter.java]: Actually, you probably want `new StringBuilder(in.length() * 2)`

File: src/camelsnake/Converter.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        DIGIT,
+        DASH,
+        UNDERSCORE,
+        OTHER
+    }
+
+    static private Type getCharType(char ch) {
+        if (Character.isUpperCase(ch)) {
+            return Type.UPPER;
+        } else if (Character.isLowerCase(ch)) {
```
Comment (jentfoo) [inline src/camelsnake/Converter.java]: I assume lower case is most common.  If you want to get every possible cycle, you could order it as the first if statement check.

**Comment by skazhy** [comment]
This is already in master. Closing.

**Pull request closed without merging**